README.md

```diff
@@ -1,2 +1,133 @@
+`~/.nce` cache. For this reason and others, you may want to ship a scie without all the files in its
+base filled in, but instead downloaded to the end users machine only if needed. This saves
+storage space on your servers and the end user's machine as well as bandwidth transferring your
+application scie between them. The `ptex` binary can help you do this.
+
+## How `ptex` works
+The [`scie-jump`](
+https://github.com/a-scie/jump/blob/main/docs/packaging.md#optional-fields) supports assigning a
+`source` binding command to any file entry in the lift manifest. That source binding command then
+becomes responsible for providing the content of that file when needed. The `ptex` binary can be
+used as the executable for these binding commands to provide a robust, self-contained means of
+fetching scie files just in time, on an as-needed basis. The `ptex` just needs to be told what URL
+each file it is responsible for materializing comes from. Configuration looks like so:
+```json
+{
+  "scie": {
+    "lift": {
+      "name": "skinny-scie",
+      "files": [
+        {
+          "name": "ptex-linux-x86_64",
+          "key": "ptex",
+          "executable": true
+        },
+        {
+          "name": "cpython-3.10.8+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz",
+          "key": "cpython",
+          "size": 27555856,
+          "hash": "6c8db44ae0e18e320320bbaaafd2d69cde8bfea171ae2d651b7993d1396260b7",
+          "type": "tar.gz",
+          "source": "ptex-fetch"
+        }
+      ],
+      "boot": {
+        "commands": {
+          "": {
+            "exe": "{cpython}/python/bin/python3.10",
+            "args": [
+              "-c",
+              "print('Hello World!')"
+            ]
+          }
+        },
+        "bindings": {
+          "ptex-fetch": {
+            "exe": "{ptex}",
+            "args": [
+              "{scie.lift}"
+            ]
+          }
+        }
+      }
+    }
+  },
+  "ptex": {
+    "cpython-3.10.8+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz":
+      "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.10.8+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz"
+  }
+}
+```
+
+In this example Linux x86_64 CPython 3.10 scie, the Linux x86_64 `ptex` binary found in the
+[`ptex` Releases](https://github.com/a-scie/ptex/releases) is directly included. The CPython binary
+distribution from [Python Build Standalone](https://github.com/indygreg/python-build-standalone) is
+not directly included, but instead sourced via `ptex`. This is arranged with a binding command
+configured to run `ptex` passing in the lift manifest itself which `ptex` uses to find the URL of
+the CPython binary distribution when it needs to fetch it.
+
+This results in a `skinny-scie` that is ~5.6MB. On 1st run on the target host you'll see some
+information about the download updating on stderr:
```

Comment:
Why `stderr`?

Reply:
`stderr` makes sense to me, that's where logs normally go.

Reply:
Yeah. If you think about pipes in shells it all makes sense. Those pipe stdout by default and the reason you can do interesting things with them is the protocol that programs produce meaningful output on stdout and output for humans on stderr. Stock curl does this with download progress info - stderr; that's why you can `curl foo > foo` and get foo the file with the meaningful data in it while still having your brain entertained by the progress output for humans at the same time.

Reply:
Fair points. Last night while reading, I had it in my head that the downloading process was a bit more fundamental to the output - but yeah, after reading both your comments - it's basically just logs.